### PR TITLE
FT: getObject with tagging

### PR DESCRIPTION
**Pull request by nicolas2bert**
E2E passed with https://github.com/scality/Integration/pull/523

Testing getObject with tagging:

- if `s3:GetObject` and `s3:GetObjectTagging` permissions : getObject return the header: `x-amz-tagging-count` that provides the count of number of tags associated with the object.

- if `s3:GetObjectVersion` and `s3:GetObjectVersionTagging` permissions : getObject with versioning return the header: `x-amz-tagging-count`.

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 1]
```diff
@@ -130,5 +130,65 @@ describe('get behavior after delete with versioning', () => {
...
+                    return done();
+                });
+            });
+
+            describe('tag associated with the object ', () => {
+                beforeEach(done => s3.putObjectTagging(paramsTagging, done));
+
+                it('should return "x-amz-tagging-count" header that provides ' +
+                'the count of number of tags associated with the object',
+                function ifF(done) {
```
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: Why `function ifF`?

File: lib/api/api.js [hunk 2]
```diff
@@ -117,7 +132,11 @@ const api = {
                 if (apiMethod === 'objectCopy' ||
                     apiMethod === 'objectPutCopyPart') {
                     return this[apiMethod](userInfo, request, sourceBucket,
-                        sourceObject, sourceVersionId, log, callback);
+                      sourceObject, sourceVersionId, log, callback);
```
Comment (bennettbuchanan) [inline lib/api/api.js]: Incredibly minor: indent?

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js (hunk 1 above)
Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: I named the it callback function `ifF`

File: lib/api/api.js (hunk 2 above)
Comment (nicolas2bert) [inline lib/api/api.js]: Thanks :)

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js (hunk 1 above)
Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: I was reading this as containing some sort of conditional logic because of the 'if'

Comment (bennettbuchanan) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: I also see `function itF` in the file.

Comment (nicolas2bert) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: Oh yea, good catch, I misspelled it. I am changing `ifF` to `itF`.

File: lib/api/api.js [hunk 3]
```diff
@@ -48,6 +48,7 @@ auth.setHandler(vault);
 /* eslint-disable no-param-reassign */
 const api = {
     callApiMethod(apiMethod, request, response, log, callback) {
+        let isHeaderTagCount = true;
```
Comment (LaurenSpiegel) [inline lib/api/api.js]: I would call this 'returnTagCount'

File: tests/functional/aws-node-sdk/test/versioning/objectGet.js [hunk 4]
```diff
@@ -130,5 +130,65 @@ describe('get behavior after delete with versioning', () => {
...
+                        return done(err);
+                    }
+                    this.currentTest.versionId = data.VersionId;
+                    return done();
+                });
+            });
+
+            it('should not return "x-amz-tagging-count" if no tag ' +
+            'associated with the object',
+            function ifF(done) {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/versioning/objectGet.js]: itF?

**Comment by nicolas2bert** [comment]
E2E with scality/Integration#523 http://ci.ironmann.io/gh/scality/Integration/13074

**Pull request merged**